### PR TITLE
[Refactor]-UI Meshery AppBar component Created

### PR DESCRIPTION
**Pull request by subhradwip01**
Signed-off-by: subhradwip01 <subhradwip01@gmail.com>

**Description**
AppBar created for meshery pattern configurator
This PR fixes #

**Notes for Reviewers**

Screenshot from 2022-08-06 10-53-29

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by YashKamboj — CHANGES_REQUESTED** [review]
@subhradwip01  Move the patternConfiguaratior folder inside the patterns folder

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+  //       );
+  //     }
+  //     setActivePatternWithRefinedSchema(activeSchema);
+  //   }
+  // }, [activeCR]);
+
+  const groupWlByVersion = () => {
+    // const mfw = meshWorkloads[selectedMeshType];
+    // return mfw ? groupWorkloadByVersion(mfw) : {};
+  }
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:105]: Can we remove this function as it does not have any purpose 👀

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+      name: name,
+      type: action
+    });
+    setSelectedPattern(resetSelectedPattern());
+  };
+
+  const setActivePatternWithRefinedSchema = async (schema) => {
+    // const refinedSchema = await getWorkloadTraitAndType(schema);
+    // setActiveForm(refinedSchema);
+  };
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:137]: This one too

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+            )}
+
+          <div>
+            <Tooltip title="Save Pattern as New File">
+              <IconButton
+                aria-label="Save"
+                color="primary"
+                // onClick={() =>
+                //   handleSubmitFinalPattern(yaml, "", getRandomName(), "upload")
+                // }
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:309]:
```suggestion
```

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+                aria-label="Update"
+                color="primary"
+                // onClick={() =>
+                //   handleSubmitFinalPattern(
+                //     yaml,
+                //     pattern.id,
+                //     pattern.name,
+                //     "update"
+                //   )
+                // }
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:325]:
```suggestion
```

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+                aria-label="Delete"
+                color="primary"
+                // onClick={() =>
+                //   handleSubmitFinalPattern(
+                //     yaml,
+                //     pattern.id,
+                //     pattern.name,
+                //     "delete"
+                //   )
+                // }
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:341]:
```suggestion
```

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+  //       );
+  //     } else {
+  //       activeSchema = selectedVersionMesh?.[selectedVersion].find(
+  //         (schema) =>
+  //           schema?.workload?.oam_definition?.metadata?.name === activeCR.name
+  //       );
+  //     }
+  //     setActivePatternWithRefinedSchema(activeSchema);
+  //   }
+  // }, [activeCR]);
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:100]: Can we remove this unused useEffect?

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+  backgroundColor: "#fff",
+  borderRadius: "8px"
+}));
+
+const PatternConfigurator = ({
+  pattern,
+  onSubmit,
+  show: setSelectedPattern
+}) => {
+  // const { workloadTraitSet, meshWorkloads } = useContext(SchemaContext);
```
Comment (adithyaakrishna) [inline ui/components/PatternConfigurator/PatternConfigurator.js:34]:
```suggestion
```

**Review by adithyaakrishna — CHANGES_REQUESTED** [review]
@subhradwip01 Could you please address the above changes? 👀

**Comment by subhradwip01** [comment]
@adithyaakrishna those functions and useEffect will be used after completing the page... some buttons and component are using those functions... if i remove this it will cause error

**Comment by subhradwip01** [comment]
> @subhradwip01 Move the patternConfiguaratior folder inside the patterns folder

Ok... should i also move other files to patterns folder from PatternConfigurator folder

**Review by YashKamboj — CHANGES_REQUESTED** [review]
@subhradwip01 I tried running your pr, it is showing this error
Screenshot (216)

**Comment by YashKamboj** [comment]
@subhradwip01 this pr is full of errors (don't implement all the functionality , just make a UI skeleton for now)

File: ui/components/PatternConfigurator/PatternConfigurator.js [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+import NameToIcon from "./NameToIcon";
+import {getMeshProperties} from "../../utils/nameMapper"
+import { isEmptyObj } from "../../utils/utils";
+import { getHumanReadablePatternServiceName, getPatternServiceName } from "./helpers";
+
+const PattrenBar = styled(AppBar)(() => ({
+  marginBottom: "16px",
+  backgroundColor: "#fff",
+  borderRadius: "8px"
+}));
```
Comment (YashKamboj) [inline ui/components/PatternConfigurator/PatternConfigurator.js:27]: Use `theme.spacing` instead of `px` and `theme.palette` to define the white color

**Review by YashKamboj — CHANGES_REQUESTED** [review]

**Comment by subhradwip01** [comment]
@YashKamboj once check it.. If any changes need to be done, tell me

File: ui/components/Patterns/PatternConfigurator.js [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+  getHumanReadablePatternServiceName,
+  getPatternServiceName,
+  isEmptyObj,
+  NameToIcon,
+  getMeshProperties,
+} from "./helpers";
+
+const PattrenBar = styled(AppBar)(({theme}) => ({
+  marginBottom: theme.spacing(2),
+  backgroundColor: theme.palette(white),
```
Comment (YashKamboj) [inline ui/components/Patterns/PatternConfigurator.js:28]: @subhradwip01  there is an error here , its a syntax error, change  it to `theme.palette.white`

**Review by YashKamboj — CHANGES_REQUESTED** [review]
@subhradwip01 And also make another folder inside patterns folder named `PatternsConfigurator` and add the `helpers.js` and `PatternConfigurator.js` in the folder

File: ui/components/Patterns/PatternConfigurator.js (hunk 9 above)
Comment (subhradwip01) [reply ui/components/Patterns/PatternConfigurator.js:28]: Ok got it...

**Comment by subhradwip01** [comment]
@YashKamboj once check it... sorry for the delay

File: ui/components/Patterns/MesheryPatterns.js [hunk 10]
```diff
@@ -1,226 +1,324 @@
...
+    /**  @type {TypeView} */
+    ("grid")
+  );
 
+  function resetSelectedRowData() {
+    return () => {
+      setSelectedRowData(null);
+    };
+  }
+  function handleSubmit({ data, id, name, type }) {
```
Comment (YashKamboj) [inline ui/components/Patterns/MesheryPatterns.js:58]: remove this function for now @subhradwip01

File: ui/components/Patterns/MesheryPatterns.js [hunk 11]
```diff
@@ -1,226 +1,324 @@
...
+          console.log("PatternFile API", `/api/pattern`);
+          updateProgress({ showProgress: false });
+          fetchPatterns(page, pageSize, search, sortOrder);
+        },
+        handleError(ACTION_TYPES.UPLOAD_PATTERN)
       );
     }
+  }
+
+  function fetchpatterns(page, pageSize, search, sortOrder) {
```
Comment (YashKamboj) [inline ui/components/Patterns/MesheryPatterns.js:125]: Remove this too

**Comment by YashKamboj** [comment]
@subhradwip01 can you fetch the latest

**Comment by Abhishek-kumar09** [comment]
What's the status here @subhradwip01 @YashKamboj

**Comment by subhradwip01** [comment]
@Abhishek-kumar09 I am waiting for the review.. already updated the required changes

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**